libs/@guardian/libs/src/storage/storage.ts

```diff
@@ -98,6 +98,38 @@ class StorageFactory {
 	setRaw(key: string, value: string): void {
 		return this.#storage?.setItem(key, value);
 	}
+
+	/**
+	 * Get all of the keys in storage.
+	 *
+	 * @returns an array of all keys in the storage object,
+	 * or undefined if storage is not available.
+	 *
+	 * @note This is not a method in the native API, and gets
+	 * all keys by iterating over the entire storage object.
+	 * It is therefore potentially subject to race conditions
+	 * and if other code is modifying storage at the same time
+	 * then the results may be inaccurate.
```

Comment:
@tjmw reminded me that JavaScript ensure run-to-completion so if the methods are synchronous it's unlikely that such a race condition may occur.

Reply:
This is a good point! Although because the storage objects are browser features I wasn't sure if we could guarantee that the storage side of the operation would be implemented synchronously..